app.js

```diff
@@ -29,33 +30,59 @@ const sessionParms = {
 };
 
 if (app.get("env") === "production") {
-  app.set("trust proxy", 1); // trust first proxy
-  sessionParms.cookie.secure = true; // serve secure cookies
+  app.set("trust proxy", 1);
+  sessionParms.cookie.secure = true;
 }
 
+app.use(cookieParser()); // Must be before CSRF middleware
 app.use(session(sessionParms));
 
 const passport = require("passport");
 const passportInit = require("./passport/passportInit");
-
 passportInit();
 app.use(passport.initialize());
 app.use(passport.session());
 
 app.use(require("connect-flash")());
-
 app.use(require("./middleware/storeLocals"));
+
+//  CSRF Protection Middleware (Must be after session and body-parser)
+app.use(csurf());
+
+//  Middleware to pass CSRF token to views
+app.use((req, res, next) => {
+  res.locals._csrf = req.csrfToken();  // Make sure CSRF token is available in views
+  next();
+});
+
+// Jobs - Add New Job Page
+app.get("/jobs/new", (req, res) => {
+  res.render("job", { job: null, _csrf: req.csrfToken() }); // Explicitly pass job as null
+});
+
+app.get("/jobs/:id/edit", async (req, res) => {
+  const job = await Job.findById(req.params.id);  // Find the job by ID
+  if (!job) {
+    return res.status(404).send("Job not found");  // Handle case when job is not found
+  }
+  res.render("job", { job, _csrf: req.csrfToken() });  // Pass job object for editing
+});
+
 app.get("/", (req, res) => {
   res.render("index");
```

Comment:
The previously mentioned host-csrf package sets this by default, so you'd only need to set your csrf token once, which could be done when a user (not necessarily signed in) hits your homepage:
```js
app.get("/", (req, res) => {
  // sets to _csrf by default
  csrf.token(req, res);
  res.render("index");
}
```
(and you could situationally refresh, like when the user logs in, as recommended in the documentation)

You'd then be able to remove your extra middleware:
```js
app.use((req, res, next) => {
  res.locals._csrf = req.csrfToken();
// ...
```
(this can go away if you switch over!)
as well as the `_csrf` passes to the `res.render()` calls.